src/shader/ShaderDynamic.js

```diff
@@ -742,19 +742,33 @@ x3dom.shader.DynamicShader.prototype.generateFragmentShader = function ( gl, pro
         }
     }
 
-    //Fog
-    if ( properties.FOG )
-    {
-        shader += x3dom.shader.fog();
-    }
-
-    if ( properties.LIGHTS || properties.CLIPPLANES )
+    // //Fog
+    // if ( properties.FOG )
+    // {
+    //     shader += x3dom.shader.fog();
+    // }
+
+    // if ( properties.LIGHTS || properties.CLIPPLANES )
+    // {
+    //     shader += "varying vec4 fragPosition;\n";
+    //     shader += "varying vec4 fragPositionWS;\n";
+    //     shader += "uniform float isOrthoView;\n";
+    // }
+
+    // same as vertex shader but with fragPositionWS for fogNoise (w/ or w/out lights)
+    if ( properties.LIGHTS || properties.FOG || properties.CLIPPLANES )
     {
+        shader += "uniform vec3 eyePosition;\n";
         shader += "varying vec4 fragPosition;\n";
-        shader += "varying vec4 fragPositionWS;\n";
         shader += "uniform float isOrthoView;\n";
-    }
+        shader += "varying vec4 fragPositionWS;\n";
 
+        if ( properties.FOG )
```

Comment:
seems cleaner to keep outside and not to nest inside here.

Reply:
originally I nested this check to mimic the exact logic used to create the corresponding vertex shader, but looks fine to moved it back outside.